### PR TITLE
BLOM-302: Update to node 14

### PR DESCRIPTION
**Pull request by JaceHensley**
## Ultimate Problem
This uses node 10 and that is leaving maintenance mode in April

## Solution
- Update to node 14
- Update pg
- Update eth/web3 related libraries

### CODE REVIEW

File: README.md
Comment (edhedges) [inline README.md:360]: Good find! Does anything else need to change for this?

File: index.ts
Comment (edhedges) [inline index.ts:98]: Does this set `stack` to `error.stack` if present, but default to `Stack is missing` if not? Pretty cute little piece of code.

File: migrations.ts
Comment (edhedges) [inline migrations.ts:183]: What's going on here?

File: src/routes/auth.ts
Comment (edhedges) [inline src/routes/auth.ts:82]: Can you get rid of or conditionally log these? We don't want these `value`s in our logs.

File: src/utils.ts
Comment (edhedges) [inline src/utils.ts:144]: Do we need to specify the encoding with Buffer.from?

File: src/utils.ts
Comment (edhedges) [inline src/utils.ts:151]: Same comment as above.

**Review by edhedges — CHANGES_REQUESTED** [review]
only a minor change needed, but overall looks good

merging will kick off the deployment so when you want to do that, let's sync backup and we can monitor it together

File: index.ts
Comment (JaceHensley) [reply index.ts:98]: Yeah, exactly!

File: README.md
Comment (JaceHensley) [reply README.md:360]: I think we could replace POSTGRES_DATABASE with POSTGRES_DB everywhere but that would require updates to the env in all the deployed locations and CI/CD

File: migrations.ts
Comment (JaceHensley) [reply migrations.ts:183]: `module.parent` is deprecated. `module.parent` would be defined if another file imported/required this file, otherwise if it was run directly (`node migrations.ts`) it would be undefined. But now you have to compare the `main` to the current module and if they are the same then that means the file was run directly

File: src/routes/auth.ts
Comment (JaceHensley) [reply src/routes/auth.ts:82]: Good catch!

File: src/utils.ts
Comment (JaceHensley) [reply src/utils.ts:144]: Nope, before toBuffer would just do `Buffer.from()` without the encoding if the passed string didn't start with `0x`

File: src/utils.ts
Comment (JaceHensley) [reply src/utils.ts:151]: Same answer as above 😜

**Review by edhedges — APPROVED** [review]
Approved! Let's wait for our meeting to merge and thus deploy so we can properly monitor / test before PRing to main and merging it in there for deployment, monitoring, etc.

**Pull request merged**
